src/libcmd/installables.cc

```diff
@@ -171,14 +171,44 @@ Strings SourceExprCommand::getDefaultFlakeAttrPathPrefixes()
 
 void SourceExprCommand::completeInstallable(std::string_view prefix)
 {
-    if (file) return; // FIXME
-
-    completeFlakeRefWithFragment(
-        getEvalState(),
-        lockFlags,
-        getDefaultFlakeAttrPathPrefixes(),
-        getDefaultFlakeAttrPaths(),
-        prefix);
+    if (file) {
+        evalSettings.pureEval = false;
+        auto state = getEvalState();
+        Expr *e = state->parseExprFromFile(
+            resolveExprPath(state->checkSourcePath(lookupFileArg(*state, *file)))
+        );
+
+        Value root;
+        state->eval(e, root);
+
+        auto autoArgs = getAutoArgs(*state);
+
+        std::string prefix_ = std::string(prefix);
+        auto sep = prefix_.rfind('.');
+        if (sep != std::string::npos) {
+            prefix_.erase(sep);
+        } else {
+            prefix_ = "";
+        }
+
+        Value &v1(*findAlongAttrPath(*state, prefix_, *autoArgs, root).first);
+        state->forceValue(v1);
+        Value v2;
+        state->autoCallFunction(*autoArgs, v1, v2);
+
+        if (v2.type() == nAttrs) {
+            for (auto & i : *v2.attrs) {
+                completions->add(i.name);
```

Comment:
This needs to filter out names that don't start with `prefix`. Otherwise I get:
```
# nix build -f ../nixpkgs hel<TAB>
Display all 14279 possibilities? (y or n)
```

It may be possible to factor out some commonality with `NixRepl::completePrefix`.

Reply:
Ahhh, `bash` doesn't filter it out on its own? I have to admit I only tested it with `zsh` which does this :) 
Thanks for pointing it out, gotta fix this!

Reply:
OK, fixed!